packages/internal/src/utils.js

```diff
@@ -348,3 +349,104 @@ export const allValues = async obj => {
   // @ts-expect-error cast
   return harden(fromEntries(zip(keys(obj), resolved)));
 };
+
+/**
+ * A tee implementation where all readers are synchronized with each other.
+ * They all consume the source stream in lockstep, and any one returning or
+ * throwing early will affect the others.
+ *
+ * @template [T=unknown]
+ * @param {AsyncIterator<T, void, void>} sourceStream
+ * @param {number} readerCount
+ */
+export const synchronizedTee = (sourceStream, readerCount) => {
```

Comment:
`synchronized` is an unnecessary qualifier. All streams are synchronized by default and can be made resilient to bursts by priming a buffer to each individual consumer. In the context of the `@endo/stream` library, `makePipe` returns a synchronized reader/writer which can be buffered by pre-acking a certain number of iterations, effectively allowing the producer to get exactly that far ahead of the consumer. Imagine:

```js
AsyncIterator.prototype.tee(quantity: length): Array<AsyncIterator>;
AsyncIterator.prototype.buffer(length: number): AsyncIterator;
```

Reply:
I think there can be non synchronized teeing. In fact I have a similar tee helper in another branch where the source is consumed at the pace of the fastest branch (allowing for unbounded memory growth).

Buffering is not equivalent as by definition that only allows for a certain bounded amount of deviation.

Reply:
I’d name these to encourage the use of bounded memory and probably wouldn’t even suggest standardizing an unbounded variant with the exception of terminal collectors like `Array.fromAsync` or `response.json()`.